### ISSUE TITLE
Problems when fusing current limiting based on istio

### ISSUE THREAD

**Issue by Aaron-23** [istio/istio#36547]
![image](https://user-images.githubusercontent.com/50609288/146502481-8001f71c-a7b8-4227-b8d8-009a9c4f363f.png)

The above is my service svc yaml file. When I define the name of `tcp-80` in spec.ports.name in the yaml file, the fuse current limit is performed, and the test is performed using fortio. The final result is as follows

![image](https://user-images.githubusercontent.com/50609288/146502537-4ccd59aa-ff8a-46bb-90e3-936934f078b7.png)

The error rate of the -1 status code in the result is 66.7%

When I define the name in the spec.ports.name field as a name that does not contain the `tcp` field, the result is normal

![image](https://user-images.githubusercontent.com/50609288/146502608-5196b6e6-262e-46c5-b6a8-06ef9ed459ef.png)

In summary, there are two problems:
1. What is the meaning of the -1 status code in the result?
2. If the spec.ports.name field contains the `tcp` field, does it conflict with some places?

**Comment by bianpengyuan** [comment]
What does `fuse current limit` mean? Do you have some configuration like fault injection enabled?

-1 means TCP connection failed, so no http response code available for the request.

tcp in the port name would affect how Istio select protocol for a port: https://istio.io/latest/docs/ops/configuration/traffic-management/protocol-selection/
